### PR TITLE
Add missing STM322U5 variants

### PR DESCRIPTION
**Pull request by noppej**
- Add support for STM32Uxx new variants, specifically 35, 45, 95, A5, 99, and A9.
- The svd file that was there for the current 75 and 85 variants also have new version available.
- Ultimately I want to make sure the STM32U5A9 chip, and it's GFXMMU peripheral work correctly, but I wanted to get the basics (above checklist) right first, and it didn't seem right to only add the A9 variant, when so many others were missing. Again, if you prefer, I can reduce this submission to only affect the A9 chip.

### Notes:
I followed these steps from the README, under the heading "Adding New Devices"
- [x] Update SVD zips in `svd/vendor` to include new SVDs.
- [x] Run `make extract` to extract the new zip files.
-[ x] Add new YAML file in `devices/` with the new SVD path and include any
  required SVD patches for this device, such as renaming or merging fields.
- [x] Add the new devices to `stm32_part_table.yaml`.
- [x] Add the new devices to `scripts/makecrates.py`.
- [ ] You can run `scripts/matchperipherals.py` script to find out what existing
  peripherals could be cleanly applied to this new SVD. If they look sensible,
  you can include them in your device YAML. This requires a Python environment with the `pyyaml`
  and `svdtools` dependencies.
  Example command: `python scripts/matchperipherals.py peripherals/rcc devices/stm32h562.yaml`
- [x] Re-run `scripts/makecrates.py devices/` to update the crates with the new devices.
- [x] Run `make` to rebuild, which will make a patched SVD and then run `svd2rust`
  on it to generate the final library.

### CODE REVIEW

File: svd/extract.sh [hunk 1]
```diff
@@ -37,8 +37,4 @@ mv stm32wle5_cm4.svd stm32wle5.svd
 
 # Copy L4X2 svd into L412; a 412 one doesn't exist.
 # We handle its modified RTC register block in `devices/stm32l412.yaml`.
-cp stm32l4x2.svd stm32l412.svd
-
-# Duplicate U5X5 to provide a copy for each chip
-cp stm32u5xx.svd stm32u575.svd
-mv stm32u5xx.svd stm32u585.svd
+cp stm32l4x2.svd stm32l412.svd
```
Comment (burrbull) [inline svd/extract.sh:40]: similar hack in `extract.ds`

**Comment by burrbull** [comment]
all file names should be lowercase

File: svd/extract.sh (hunk 1 above)
Comment (noppej) [reply svd/extract.sh:40]: Thanks ... appreciate the help.

**Comment by noppej** [comment]
@burrbull I fumbled a bit with fixing all the lowercase instances of the device files, but it still fails with 
```
warning: failed to load external entity "svd/stm32u5A5.svd.patched"
warning: failed to load external entity "svd/stm32u5A9.svd.patched"
```

Do have any pointers for me on where I'm messing that up?

**Comment by burrbull** [comment]
`stm32u5A5.yaml` -> `stm32u5a5.yaml`

**Comment by noppej** [comment]
> `stm32u5A5.yaml` -> `stm32u5a5.yaml`, same for `a9`

When I look at my `/devices' folder, this is what I see ... they are all lowercase.
```
 % ls -la stm32u5*
-rw-r--r--  1 jackn  staff  49 23 Sep 13:22 stm32u535.yaml
-rw-r--r--  1 jackn  staff  49 23 Sep 13:22 stm32u545.yaml
-rw-r--r--  1 jackn  staff  49 23 Sep 13:22 stm32u575.yaml
-rw-r--r--  1 jackn  staff  49 23 Sep 13:22 stm32u585.yaml
-rw-r--r--  1 jackn  staff  49 23 Sep 13:22 stm32u595.yaml
-rw-r--r--  1 jackn  staff  49 23 Sep 13:22 stm32u599.yaml
-rw-r--r--  1 jackn  staff  49 23 Sep 14:46 stm32u5a5.yaml
-rw-r--r--  1 jackn  staff  49 23 Sep 14:46 stm32u5a9.yaml
```

**Comment by noppej** [comment]
@burrbull I am sorry to be a pain. I've managed to overcome the issue above, but now something else that seems weird. 

The `compare mmaps` check fails, because it can't find `stm32u535.svd.patched` . However, if I manually run the `make patch` on my local system, I end up with these files.
```
% ls -la stm32u535*
-rw-rw-r--  1 jackn  staff  5891707  9 Mar  2023 stm32u535.svd
-rw-r--r--  1 jackn  staff  4675187 23 Sep 18:59 stm32u535.svd.patched
```

**Comment by burrbull** [comment]
> @burrbull I am sorry to be a pain. I've managed to overcome the issue above, but now something else that seems weird.
> 
> The `compare mmaps` check fails, because it can't find `stm32u535.svd.patched` . However, if I manually run the `make patch` on my local system, I end up with these files.
> 
> ```
> % ls -la stm32u535*
> -rw-rw-r--  1 jackn  staff  5891707  9 Mar  2023 stm32u535.svd
> -rw-r--r--  1 jackn  staff  4675187 23 Sep 18:59 stm32u535.svd.patched
> ```

It's normal. CI runs `master` version of `mmaps` to compare. It doesn't know about added/removed files, only changed.
So @adamgreig should first merge this PR manually to fix CI.

Squash commits, please.

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,6 @@ mmaps/
 
 # automatically generated crates
 /stm32*/
+
+# It's a Mac Thing
+**/.DS_Store
```
Comment (adamgreig) [inline .gitignore:17]:
```suggestion
```

You can put things like this in a global `.gitignore` (e.g. in your home directory), rather than having to write it in every project: `git config --global core.excludesfile ~/.gitignore` for instance. I think it should be enough to write `.DS_Store` without the leading `**/` too, fwiw.

File: README.md [hunk 3]
```diff
@@ -89,34 +90,33 @@ features = ["stm32f405", "rt"]
 The nightlies should always build and be as stable as the latest release, but
 contain the latest patches and updates.
 
-
 ## Generating Device Crates / Building Locally
 
-* Install `svd2rust`, `svdtools`, and `form`:
```
Comment (adamgreig) [inline README.md:95]: Sorry to be a pain, but could you undo these changes? I guess you have something auto-format markdown files, but this PR doesn't need to change all the bullet point lists!

**Review by adamgreig — CHANGES_REQUESTED** [review]
Thank you for the PR! The U5 changes look good but I had a couple of comments about the unrelated changes:

File: README.md (hunk 3 above)
Comment (noppej) [reply README.md:95]: Totally reasonable. I will undo :)

**Review by adamgreig — APPROVED** [review]

**Comment by adamgreig** [comment]
perfect, thanks very much!

**Comment by noppej** [comment]
> perfect, thanks very much!

You're welcome. Thanks to you and @burrbull for your patience as I get up to speed with this repo.

**Comment by noppej** [comment]
Thanks.  I will look at it and update the PR.  I downloaded it when I started working on it so maybe I missed an update.
________________________________
